random failure of core package testing

On October 31st, I found that the test cases of the core package would randomly fail.

```shell
go test ./core/...
```

Sometimes it fails:

```text
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xb0 pc=0x71406c]

goroutine 8644 [running]:
github.com/XinFinOrg/XDPoSChain/core/state.(*StateDB).Finalise(0x0, 0x1)
        /home/me/XDPoSChain/core/state/statedb.go:771 +0x2c
github.com/XinFinOrg/XDPoSChain/core/state.(*StateDB).IntermediateRoot(0x0, 0x0?)
        /home/me/XDPoSChain/core/state/statedb.go:804 +0x4b
github.com/XinFinOrg/XDPoSChain/core.(*statePrefetcher).Prefetch(0xc001893800, 0xc0016d07e0, 0x0, {0x0, 0x0, 0x0, {0x0, 0x0, 0x0}}, 0xc0007f2a50)
        /home/me/XDPoSChain/core/state_prefetcher.go:82 +0x58f
github.com/XinFinOrg/XDPoSChain/core.(*BlockChain).insertChain.func1({0x8d47e5?, 0x4263a5?, 0x11ad8e0?})
        /home/me/XDPoSChain/core/blockchain.go:1778 +0x116
created by github.com/XinFinOrg/XDPoSChain/core.(*BlockChain).insertChain in goroutine 7697
        /home/me/XDPoSChain/core/blockchain.go:1773 +0x1318
FAIL    github.com/XinFinOrg/XDPoSChain/core    12.258s
```

After troubleshoot, it was caused by the PR #1662:

```text
2025-10-31 12:26:22 336d1f9f core: fix too many goroutines in InitSignerInTransactions, close XFN-75 (#1662)
2025-10-31 12:24:32 a5e68b9c miner: not sleep if locks multiple mutex, close XFN-40 (#1639)
```

Use the following script to test commits `336d1f9f ` and `a5e68b9c`:

```shell
COUNTER=1
while true; do
    echo
    echo "$(date '+%Y-%m-%d %H:%M:%S'): test no ${COUNTER}"
    go clean -testcache
    go test -v ./core -run TestDAOForkRangeExtradata
    
    if [[ $? -ne 0 ]]; then
        echo "test failed"
        break
    fi   
    ((COUNTER++))
    sleep 1
done
```

The commits `336d1f9f ` will break test quickly, usually within 10 times:

result:

```text
2025-11-05 11:28:23: test no 1
=== RUN   TestDAOForkRangeExtradata
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xb0 pc=0x71406c]

goroutine 1000 [running]:
github.com/XinFinOrg/XDPoSChain/core/state.(*StateDB).Finalise(0x0, 0x1)
        /home/me/XDPoSChain/core/state/statedb.go:771 +0x2c
github.com/XinFinOrg/XDPoSChain/core/state.(*StateDB).IntermediateRoot(0x0, 0x0?)
        /home/me/XDPoSChain/core/state/statedb.go:804 +0x4b
github.com/XinFinOrg/XDPoSChain/core.(*statePrefetcher).Prefetch(0xc001227600, 0xc00119f830, 0x0, {0x0, 0x0, 0x0, {0x0, 0x0, 0x0}}, 0xc001482788)
        /home/me/XDPoSChain/core/state_prefetcher.go:82 +0x58f
github.com/XinFinOrg/XDPoSChain/core.(*BlockChain).insertChain.func1({0xc0002207d0?, 0x4263a5?, 0x11ad8e0?})
        /home/me/XDPoSChain/core/blockchain.go:1778 +0x116
created by github.com/XinFinOrg/XDPoSChain/core.(*BlockChain).insertChain in goroutine 33
        /home/me/XDPoSChain/core/blockchain.go:1773 +0x1318
FAIL    github.com/XinFinOrg/XDPoSChain/core    0.079s
FAIL
test failed
```

The commit `a5e68b9c` has no error after man loop times.

The PR #1742 fixes this issue. 

The PR #1719 can also fixes this issue by disable prefetch.

--- a/core/blockchain.go
+++ b/core/blockchain.go
@@ -1781,18 +1781,19 @@ func (bc *BlockChain) insertChain(chain types.Blocks, verifySeals bool) (int, []
 		var followupInterrupt atomic.Bool
 		if !bc.cacheConfig.TrieCleanNoPrefetch {
 			if followup, err := it.peek(); followup != nil && err == nil {
-				go func(start time.Time) {
-					throwaway, _ := state.New(parent.Root, bc.stateCache)
+				throwaway, _ := state.New(parent.Root, bc.stateCache)
+
+				go func(start time.Time, followup *types.Block, throwaway *state.StateDB, interrupt *atomic.Bool) {
 					// Disable tracing for prefetcher executions.
 					vmCfg := bc.vmConfig
 					vmCfg.Tracer = nil
-					bc.prefetcher.Prefetch(followup, throwaway, vmCfg, &followupInterrupt)
+					bc.prefetcher.Prefetch(followup, throwaway, vmCfg, interrupt)
 
 					blockPrefetchExecuteTimer.Update(time.Since(start))
-					if followupInterrupt.Load() {
+					if interrupt.Load() {
 						blockPrefetchInterruptMeter.Mark(1)
 					}
-				}(time.Now())
+				}(time.Now(), followup, throwaway, &followupInterrupt)
 			}
 		}
 

